io.c

```diff
@@ -1126,7 +1126,7 @@ static int io_discard(struct io_kiocb *req, const struct sqe_submit *s,
 
 	inode = req->file->f_inode;
 	if (S_ISBLK(inode->i_mode)) {
-		struct block_device *bdev = inode->i_bdev;
+		struct block_device *bdev = I_BDEV(inode);
```

Comment:
uring is not in release branch.

Reply:
i would be surprised. I_BDEV macro is in vanilla kernel from as earlier as 3.9 version (i didnt got before that).

Reply:
here.. its from ages before... that first commit has I_BDEV
commit 1da177e4c3f41524e886b7f1b8a0c1fc7321cac2 (tag: v2.6.12-rc2)
Author: Linus Torvalds <torvalds@ppc970.osdl.org>
Date:   Sat Apr 16 15:20:36 2005 -0700

    Linux-2.6.12-rc2

    Initial git repository build. I'm not bothering with the full history,
    even though we have it. We can create a separate "historical" git
    archive of that later if we want to, and in the meantime it's about
    3.2GB when imported into git - space that would just make the early
    git days unnecessarily complicated, when we don't have a lot of good
    infrastructure for it.

    Let it rip!

Reply:
:+1: